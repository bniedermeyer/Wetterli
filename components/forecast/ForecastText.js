import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default function ForecastText(props) {
  if (!props.forecast) {
    return null;
  }

  return (
    <View>
      <Text style={styles.forcastText}>{props.forecast.temperature}Â°</Text>

      <Text style={styles.forcastText}>{props.forecast.summary}</Text>
      <Text
        style={styles.darkSkyLink}
        onPress={() => Linking.openURL('https://darksky.net/poweredby')}
      >
        Powered by Dark Sky
      </Text>
    </View>
  );
}

const styles = StyleSheet.create({
  forcastText: {
    color: '#fff',
    fontSize: 30,
    textAlign: 'center'
  },
  darkSkyLink: {
    color: '#f25f4c',
    textAlign: 'center'
  }
});
