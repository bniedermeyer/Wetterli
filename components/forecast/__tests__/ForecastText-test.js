import React from 'react';
import renderer from 'react-test-renderer';

import ForecastText from '../ForecastText';

it(`renders correctly`, () => {
  const forecast = {
    apparentTemperature: 38.14,
    cloudCover: 0.69,
    dewPoint: 37.12,
    humidity: 0.88,
    icon: 'partly-cloudy-day',
    nearestStormBearing: 178,
    nearestStormDistance: 5,
    ozone: 385.9,
    precipIntensity: 0,
    precipProbability: 0,
    pressure: 1009.6,
    summary: 'Mostly Cloudy',
    temperature: 40.47,
    time: 1577038353,
    uvIndex: 0,
    visibility: 10,
    windBearing: 352,
    windGust: 5.98,
    windSpeed: 3.61
  };
  const tree = renderer.create(<ForecastText forecast={forecast} />).toJSON();

  expect(tree).toMatchSnapshot();
});
