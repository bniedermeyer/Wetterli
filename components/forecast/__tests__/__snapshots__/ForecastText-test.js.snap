// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "textAlign": "center",
      }
    }
  >
    40.47
    Â°
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 30,
        "textAlign": "center",
      }
    }
  >
    Mostly Cloudy
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#f25f4c",
        "textAlign": "center",
      }
    }
  >
    Powered by Dark Sky
  </Text>
</View>
`;
